jib-core/src/main/java/com/google/cloud/tools/jib/api/ImageReference.java

```diff
@@ -137,18 +138,11 @@ public static ImageReference parse(String reference) throws InvalidImageReferenc
       repository = LIBRARY_REPOSITORY_PREFIX + repository;
     }
 
-    if (!Strings.isNullOrEmpty(tag)) {
-      if (!Strings.isNullOrEmpty(digest)) {
-        // Cannot have matched both tag and digest.
-        throw new InvalidImageReferenceException(reference);
-      }
-    } else if (!Strings.isNullOrEmpty(digest)) {
-      tag = digest;
-    } else {
+    if (Strings.isNullOrEmpty(tag)) {
       tag = DEFAULT_TAG;
     }
 
-    return new ImageReference(registry, repository, tag);
+    return new ImageReference(registry, repository, tag, digest);
```

Comment:
I actually think we can call `of(registry, repository, tag, digest)` here and get rid of some `if` checks in this method, as `parse()` and `of()` duplicate some logic. But let's not touch this method anymore in this PR to be on the safe side.